integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/PersonFhirResourceProviderIntegrationTest.java

```diff
@@ -0,0 +1,455 @@
+		
+		assertThat(person.getAddress().get(0).getCity(), equalTo("Kampala"));
+		assertThat(person.getAddress().get(0).getState(), equalTo("Mukono"));
+		assertThat(person.getAddress().get(0).getCountry(), equalTo("Uganda"));
+		assertThat(person, validResource());
+		
+		// try to get new person
+		response = get("/Person/" + person.getIdElement().getIdPart()).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		
+		Person newPerson = readResponse(response);
+		
+		assertThat(newPerson.getId(), equalTo(person.getId()));
+	}
+	
+	@Test
+	public void shouldCreateNewPersonAsXML() throws Exception {
+		// read XML record
+		String xmlPerson;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(XML_CREATE_PERSON)) {
+			Objects.requireNonNull(is);
+			xmlPerson = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
+		
+		// create person
+		MockHttpServletResponse response = post("/Person").accept(FhirMediaTypes.XML).xmlContext(xmlPerson).go();
+		
+		// verify created correctly
+		assertThat(response, isCreated());
+		assertThat(response.getHeader("Location"), containsString("/Person/"));
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		Person person = readResponse(response);
+		
+		assertThat(person, notNullValue());
+		assertThat(person.getIdElement().getIdPart(), notNullValue());
+		assertThat(person.getName().get(0).getGiven().get(0).toString(), equalToIgnoringCase("Adam"));
+		assertThat(person.getName().get(0).getFamily(), equalToIgnoringCase("John"));
+		assertThat(person.getGender(), equalTo(Enumerations.AdministrativeGender.MALE));
+		
+		Date birthDate = Date.from(LocalDate.of(2004, 8, 12).atStartOfDay(ZoneId.systemDefault()).toInstant());
+		assertThat(person.getBirthDate(), equalTo(birthDate));
+		
+		assertThat(person.getAddress().get(0).getCity(), equalTo("Kampala"));
+		assertThat(person.getAddress().get(0).getState(), equalTo("Mukono"));
+		assertThat(person.getAddress().get(0).getCountry(), equalTo("Uganda"));
+		assertThat(person, validResource());
+		
+		// try to get new person
+		response = get("/Person/" + person.getIdElement().getIdPart()).accept(FhirMediaTypes.XML).go();
+		
+		assertThat(response, isOk());
+		
+		Person newPerson = readResponse(response);
+		
+		assertThat(newPerson.getId(), equalTo(person.getId()));
+	}
+	
+	@Test
+	public void shouldUpdateExistingPersonAsJson() throws Exception {
+		// get the existing record
+		MockHttpServletResponse response = get("/Person/" + PERSON_UUID).accept(FhirMediaTypes.JSON).go();
+		Person person = readResponse(response);
+		
+		// update the existing record
+		Date birthDate = DateUtils.truncate(new Date(), Calendar.DATE);
+		person.setBirthDate(birthDate);
+		
+		// send the update to the server
+		response = put("/Person/" + PERSON_UUID).jsonContent(toJson(person)).go();
+		
+		assertThat(response, isOk());
```

Comment:
@varung-31  the same goes here with this [error](https://pastebin.com/k1HzWgNK)  of which i think the person must be some how connect to a particular patient in the system